test-case/check-pause-release-suspend-resume.sh

```diff
@@ -112,73 +106,77 @@ logger_disabled || func_lib_start_log_collect
+
+    # expect is tcl language script
+    # catch: Evaluate script and trap exceptional returns
+    # after ms: Ms must be an integer giving a time in milliseconds.
+    # The command sleeps for ms milliseconds and then returns.
+    expect <<AUDIO
+    spawn $cmd -D $dev -r $rate -c $channel -f $fmt -vv -i $file_name -q
+    set i 1
+    set sleep_t $sleep_period
+    expect {
+        "#*+*\%" {
+            # audio stream (aplay or arecord) is active now and playing
+            puts "\r===== (\$i/$repeat_count) pb_pbm: Pause $cmd, then wait for ===== "
+            puts "\r(\$i/$repeat_count) pb_pbm: $sleep_t ms after pause"
+            send " "
+            after \$sleep_t
+            puts "Finished sleep. Confirming $cmd is paused."
+
+            # check audio stream status --- R == paused, else == not paused
+            set retval [catch { exec ps -C $cmd -o state --no-header } state]
+            puts "$cmd state = \$state"
+
+            if {[string equal \$state "R"] != 0} {
+                puts "$cmd is paused, will now enter suspend-resume cycle"
+            } else {
+                puts "$cmd is not paused. Exiting test and failing."
+                exit 1
+            }
+
+            # enter suspend-resume cycle once per pause instance
+            set retval [catch { exec bash $CASEDIR/check-suspend-resume.sh -l 1 } msg]
+
+            # prints logs from suspend-resume test
+            puts \$msg
+            puts "Finished suspend-resume test"
+            if { \$retval } {
+                puts "suspend resume cycle has failed."
+                set error_code [lindex \$::errorCode {2}]
+                puts "errorCode was: \$error_code"
+                exit 1
+            }
+
+            # sucessful suspend/resume cycle, now release audio stream
+            puts "\r(\$i/$repeat_count) pb_pbm: Release $cmd, then wait for"
+            puts "\r(\$i/$repeat_count) pb_pbm: \$sleep_t ms after resume"
+            send " "
+            after \$sleep_t
+
+            puts "Finished sleep after resume"
+            incr i
+            if { \$i > $repeat_count } { exit 0 }
+            exp_continue
         }
-
-        #sucessful suspend/resume cycle, now release audio stream
-        puts "\r(\$i/$repeat_count) pb_pbm: Release $cmd, then wait for"
-        puts "\r(\$i/$repeat_count) pb_pbm: \$sleep_t ms after resume"
-        send " "
-        after \$sleep_t
-
-        puts "Finished sleep after resume"
-        incr i
-        if { \$i > $repeat_count } { exit 0 }
-        exp_continue
     }
-}
 AUDIO
 
-ret=$?
-#flush the output
-echo
-if [ $ret -ne 0 ]; then
-    sof-process-kill.sh
-    [[ $? -ne 0 ]] && dlogw "Kill process catch error"
-    exit $ret
-fi
-sof-kernel-log-check.sh "$KERNEL_CHECKPOINT"
-exit $?
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        sof-process-kill.sh || dlogw "sof-process-kill.sh failed"
+        exit $ret
```

Comment:
Do we really want to preserve the exit code here? If it's 2 then it will be misinterpreted as a SKIP! We experienced this recently in some different place.